source/android/Renaissance/lib/edit_profile.dart

```diff
@@ -0,0 +1,217 @@
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:portakal/file_converter.dart';
+import 'package:snippet_coder_utils/multi_images_utils.dart';
+
+import 'http_services.dart';
+import 'models/user.dart';
+import 'my_colors.dart';
+
+class EditProfilePage extends StatefulWidget {
+  const EditProfilePage({Key? key, required this.delegate}) : super(key: key);
+  final EditProfileDelegate delegate;
+
+  @override
+  _EditProfilePageState createState() => _EditProfilePageState();
+}
+
+mixin EditProfileDelegate {
```

Comment:
Can you briefly explain what this mixing provides to the project? I wasn't able to figure it out.

Reply:
Mixin means protocol in Flutter. In order to let the previous page know about the changes that are done in edit profile page, we define a mixin and create an instance of it, inside Edit Profile Page. When save button is pressed and changes are made, edit profile calls the onSuccessfullSave() function via its delegate. Since Profile Page conforms to EditProfileDelegate protocol, it is instantly notified about the changes. So, when we change a user attribute and return back, profile page is also changed.